### PR TITLE
AzureCLI integration

### PR DESCRIPTION
**Pull request by seriva**
- Added login to Azure-CLI and selection of subscription.
- Added creation of service principle.
- Added creation of the resource group.
- Added possibility to run without service principal

### CODE REVIEW

File: core/src/epicli/data/common/defaults/epiphany-cluster.yml [hunk 1]
```diff
@@ -11,8 +11,10 @@ specification:
     key_path: /root/.ssh/epiphany-operations/id_rsa # YOUR-SSH-KEY-PATH
   cloud:
     subscription_name: YOUR-SUB-NAME
+    resource_group_name: YOUR-RESOURCE-GROUP-NAME
     vnet_address_pool: 10.1.0.0/20
     use_public_ips: False # When not using public IPs you have to provide connectivity via private IPs (VPN)
+    use_service_principle: False
```
Comment (erzetpe) [inline core/src/epicli/data/common/defaults/epiphany-cluster.yml]: should be probably use_service_principal

Comment (erzetpe) [reply core/src/epicli/data/common/defaults/epiphany-cluster.yml]: do we also need this for AWS?

File: core/src/epicli/data/common/defaults/epiphany-cluster.yml [hunk 2]
```diff
@@ -11,8 +11,10 @@ specification:
     key_path: /root/.ssh/epiphany-operations/id_rsa # YOUR-SSH-KEY-PATH
   cloud:
     subscription_name: YOUR-SUB-NAME
+    resource_group_name: YOUR-RESOURCE-GROUP-NAME
```
Comment (erzetpe) [inline core/src/epicli/data/common/defaults/epiphany-cluster.yml]: shouldn't this be created based on cluster name, similar to this what we have on AWS? we also are not using this parameter on AWS in terraform templates probably?

Comment (seriva) [reply core/src/epicli/data/common/defaults/epiphany-cluster.yml]: This is already configurable for AWS as well

**Pull request merged**